### PR TITLE
Add Waitlist guide; add setup instructions to waitlist component reference

### PR DESCRIPTION
**Pull request by alexisintech**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1734/components/waitlist/overview
> - https://clerk.com/docs/pr/1734/guides/waitlist

### Explanation:

- The instructions on how to set up the waitlist component was not clear.

### This PR:

- Adds a guide for setting up an app with a waitlist
- Component reference: Adds instructions for enabling Waitlist mode in the Clerk Dashboard and adds warning callout that the user must configure ClerkProvider before using the waitlist component
- Changes the URL for the component reference: `/docs/components/waitlist/overview` --> `/docs/components/waitlist`
  - PR here for the redirect: https://github.com/clerk/clerk/pull/811

### Checklist

- [X] I have clicked on "Files changed" and performed a thorough self-review
- [X] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [X] All existing checks pass

### CODE REVIEW

File: docs/guides/waitlist.mdx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+  The [`<ClerkProvider>`](/docs/components/clerk-provider) component wraps your app to provide active session and user context to Clerk's hooks and other components. To use the `<Waitlist />` component in your app, you must provide the `waitlistUrl` prop.
+
+  ```tsx {{ filename: 'app/layout.tsx', mark: [6] }}
+  import { ClerkProvider } from '@clerk/nextjs'
+  import './globals.css'
+
+  export default function RootLayout({ children }: { children: React.ReactNode }) {
+    return (
+      <ClerkProvider waitlistUrl="/waitlist">
```
Comment (nikospapcom) [inline docs/guides/waitlist.mdx:77]: @alexisintech I saw that you render the `<Waitlist />` component in `/`, so the `waitlistUrl` prop should be `/`

File: docs/guides/waitlist.mdx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+  #### Add a sign-in page
+
+  The following example demonstrates how to render the `<SignIn />` component. **Pass the `waitlistUrl` prop to the component to provide the target of the 'Waitlist' link that's rendered in the `<SignIn />` component.**
+
+  ```tsx {{ filename: 'app/sign-in/[[...sign-in]]/page.tsx' }}
+  import { SignIn } from '@clerk/nextjs'
+
+  export default function Page() {
+    return <SignIn waitlistUrl="/" />
```
Comment (nikospapcom) [inline docs/guides/waitlist.mdx:143]: If we pass the `waitlistUrl` in `<ClerkProvider>` we don't need to add it here

Comment (alexisintech) [reply docs/guides/waitlist.mdx]: addressed: [a93f692](https://github.com/clerk/clerk-docs/pull/1734/commits/a93f692050989406698140b8968afd82161215e8)

File: docs/guides/waitlist.mdx (hunk 1 above)
Comment (alexisintech) [reply docs/guides/waitlist.mdx]: fixed! [a93f692](https://github.com/clerk/clerk-docs/pull/1734/commits/a93f692050989406698140b8968afd82161215e8)

**Comment by alexisintech** [comment]
blocked until redirects are merged: https://github.com/clerk/clerk/pull/811

**Pull request merged**